build/yaml/crd/nsx.vmware.com_vpcnetworkconfigurations.yaml

```diff
@@ -0,0 +1,138 @@
+      jsonPath: .spec.PrivateCIDRs
+      name: Private CIDRs
+      type: string
+    name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: VPCNetworkConfiguration is the Schema for the vpcnetworkconfigurations
+          API.
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: VPCNetworkConfigurationSpec defines the desired state of
+              VPCNetworkConfiguration
+            properties:
+              appliedToNamespaces:
+                description: Namespaces to be applied.
+                items:
+                  type: string
+                type: array
+              defaultSubnetPrefixLength:
+                description: Default subnet prefix length for Subnet. Defaults to
+                  28.
+                type: integer
+              defaultSubnetType:
+                description: DefaultSubnetType defines the type of the default SubnetSet
+                  for PodVM and VM. Must be public or private. Defaults to public.
+                enum:
+                - public
+                - private
+                type: string
+              edgeCluster:
+                description: Edge cluster paths on which the networking elements will
+                  be created.
+                type: string
+              nsxtProject:
+                description: NSX-T Project of the Namespace associated with.
+                type: string
+              privateCIDRs:
+                description: Private CIDRs used to allocate private Subnets.
+                items:
+                  type: string
+                maxItems: 5
+                minItems: 0
+                type: array
+              publicIPV4Pools:
+                description: NSX-T IP Block paths used to allocate public Subnets.
+                items:
+                  type: string
+                maxItems: 5
+                minItems: 1
+                type: array
+              tier0Gateway:
+                description: Tier0 gateway or Tier0 VRF path.
+                type: string
+            required:
+            - edgeCluster
+            - nsxtProject
+            - publicIPV4Pools
```

Comment:
They cannot be optional to reuse cluster level configuration?

Reply:
They can inherit Supervisor level settings.
As discussed with Jianjun, we can set these fields optional in VAPI, but let wcpsvc fill these fields for us.

Reply:
I thought what we concluded is all these fields are optional in CRD too, and if they are not set we use the Supervisor level values?

Reply:
Yes.